File: go/functions/json_operations.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
)

type Person struct {
	Name   string   `json:"name"`
	Age    int      `json:"age"`
	Skills []string `json:"skills"`
	Active bool     `json:"active"`
}

func main() {
	person := Person{
		Name:   "John Doe",
		Age:    30,
		Skills: []string{"Go", "Python", "Docker"},
		Active: true,
	}

	jsonBytes, err := json.MarshalIndent(person, "", "  ")
	if err != nil {
		fmt.Printf("Error marshaling: %v\n", err)
		return
	}

	fmt.Printf("JSON String:\n%s\n", string(jsonBytes))

	var parsedPerson Person
	err = json.Unmarshal(jsonBytes, &parsedPerson)
	if err != nil {
		fmt.Printf("Error unmarshaling: %v\n", err)
		return
	}

	fmt.Printf("\nParsed data: %+v\n", parsedPerson)
	fmt.Printf("Name: %s\n", parsedPerson.Name)
	fmt.Printf("Skills: %v\n", parsedPerson.Skills)

	jsonFile := "../tmp/data.json"
	err = ioutil.WriteFile(jsonFile, jsonBytes, 0644)
	if err != nil {
		fmt.Printf("Error writing file: %v\n", err)
		return
	}
	fmt.Printf("\nWritten to %s\n", jsonFile)
}

File: python/functions/json_operations.py
#!/usr/bin/env python3

import json

data = {
    'name': 'John Doe',
    'age': 30,
    'skills': ['Python', 'JavaScript', 'SQL'],
    'active': True
}

json_string = json.dumps(data, indent=2)
print(f"JSON String:\n{json_string}")

parsed_data = json.loads(json_string)
print(f"\nParsed data: {parsed_data}")
print(f"Name: {parsed_data['name']}")
print(f"Skills: {', '.join(parsed_data['skills'])}")

json_file = '../tmp/data.json'
with open(json_file, 'w') as file_handle:
    json.dump(data, file_handle, indent=2)
print(f"\nWritten to {json_file}")

with open(json_file, 'r') as file_handle:
    loaded_data = json.load(file_handle)
print(f"Loaded from file: {loaded_data}")
